apps/system/js/advanced_telemetry.js

```diff
@@ -0,0 +1,426 @@
+        registerHandlers();
+      });
+    }
+
+    function registerHandlers() {
+      // Basic event handlers
+      EVENT_TYPES.forEach(function (type) {
+        window.addEventListener(type, self);
+      });
+
+      self.startBatch();
+
+      if (done) {
+        loginfo('Initialization Complete');
+        done();
+      }
+    }
+  };
+
+  // This method shuts everything down and is only exposed for unit testing.
+  // Note that this is not the same as the stopCollecting() method which is
+  // used to stop data collection but keep the module running.
+  AT.prototype.stop = function stop() {
+
+  };
+
+  // Reset (or initialize) the AdvancedTelemetry instance variables
+  AT.prototype.reset = function() {
+    this.collecting = false;
+    this.metrics = null;
+    this.deviceID = null;
+  };
+
+  AT.prototype.handleEvent = function handleEvent(e) {
+    switch (e.type) {
+      case HUDMETRICS:
+        this.transmit(e.detail);
+        break;
+      default:
+        break;
+    }
+  };
+
+  // Signal gecko to send the histograms with a telemetry message.
+  AT.prototype.getPayload = function getPayload() {
+    console.info('telemetry|MGMT|TIMETOSHIP');
+  };
+
+  // Start a timer to notify when to send the payload.
+  AT.prototype.startBatch = function startBatch() {
+    var self = this;
+    this.timer = setTimeout(function() {
+      if (navigator.onLine) {
+        self.getPayload();
+      }
+    }, self.metrics.getInterval());
+  };
+
+  AT.prototype.stopBatch = function stopBatch() {
+    clearTimeout(this.timer);
+  };
+
+  // Once we have the histogram payload, we can send it to the telemetry
+  // server.  Once we receive 200 OK back from telemetry server, we can
+  // start a fresh batch of metrics.
+  AT.prototype.transmit = function transmit(payload) {
+    var self = this;
+
+    if (!this.collecting) {
+      return;
+    }
+
+    // But assume that it will succeed and start collecting new metrics now
+    this.metrics = new BatchTiming(false);
+    this.startBatch();
+
+    var deviceInfoQuery = {
+      'deviceinfo.hardware': 'unknown',
+      'app.update.channel': 'unknown',
+      'deviceinfo.platform_version': 'unknown',
+      'deviceinfo.platform_build_id': 'unknown'
+    };
+
+
+    // Query the settings db for parameters for the URL
+    AT.getSettings(deviceInfoQuery, function(deviceResponse) {
+      // Note that this wrapper is using the new v4 Unified Telemetry format
+      var wrapper = {
+        type: REASON,
+        id: uuid(),
+        creationDate: new Date().toISOString(),
+        version: AT.TELEMETRY_VERSION,
+        application: {
+          architecture: 'x86',
+          buildId: deviceResponse['deviceinfo.platform_build_id'],
+          name: AT.TELEMETRY_APP_NAME,
+          version: deviceResponse['deviceinfo.platform_version'],
+          vendor: 'Mozilla',
+          platformVersion: deviceResponse['deviceinfo.platform_version'],
+          xpcomAbi: 'x86-msvc',
```

Comment:
I have to assume this is just a placeholder value, since most FxOS devices are using `armeabi` binaries. does the telemetry service care about these values for FxOS?